app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,70 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: %i[ show edit update destroy ]
+
+  # GET /comments or /comments.json
+  def index
+    @comments = Comment.all
+  end
+
+  # GET /comments/1 or /comments/1.json
+  def show
+  end
+
+  # GET /comments/new
+  def new
+    @comment = Comment.new
+  end
+
+  # GET /comments/1/edit
+  def edit
+  end
+
+  # POST /comments or /comments.json
+  def create
+    @comment = Comment.new(comment_params)
+
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to comment_url(@comment), notice: "Comment was successfully created." }
+        format.json { render :show, status: :created, location: @comment }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @comment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /comments/1 or /comments/1.json
+  def update
+    respond_to do |format|
+      if @comment.update(comment_params)
+        format.html { redirect_to comment_url(@comment), notice: "Comment was successfully updated." }
+        format.json { render :show, status: :ok, location: @comment }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @comment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /comments/1 or /comments/1.json
+  def destroy
+    @comment.destroy
+
+    respond_to do |format|
+      format.html { redirect_to comments_url, notice: "Comment was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_comment
```

Comment:
The way you load resources in the set_comment method is standard, but as your application grows, consider handling cases where a comment might not be found (e.g., Comment.find_by(id: params[:id])). You can then handle the nil case explicitly, perhaps by redirecting to the index with a flash message indicating that the comment was not found.